4.gene_expression_signatures/scripts/0.latent_dimension_dataframe.py

```diff
@@ -0,0 +1,283 @@
+sys.path.insert(0, str(script_directory))
+from data_loader import load_train_test_data, load_model_data
+
+
+# In[2]:
+
+
+# Load command line arguments
+args = get_optimize_args()
+tc_args = get_optimize_args_tc()
+vvae_args = get_optimize_args_vvae()
+
+# Load data
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+
+train_df, test_df, val_df, load_gene_stats = load_train_test_data(
+    data_directory, train_or_test="all", load_gene_stats=True, zero_one_normalize=True
+)
+train_data = pd.DataFrame(train_df)
+
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve()
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet").resolve()
+dependency_df, gene_dict_df= load_model_data(dependency_file, gene_dict_file)
+gene_dict_df = pd.DataFrame(gene_dict_df)
+
+
+# In[3]:
+
+
+#Load weight data for VAEs
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+weight_df = load_train_test_data(
+    data_directory, train_or_test="train"
+)
+
+gene_list_passed_qc = gene_dict_df.loc[
+    gene_dict_df["qc_pass"], "dependency_column"
+].tolist()
+
+weight_data = weight_df.filter(gene_list_passed_qc, axis=1)
+weight_data.head()
+
+
+# In[4]:
+
+
+# Convert dataframes to tensors
+train_tensor = torch.tensor(train_df, dtype=torch.float32)
+test_tensor = torch.tensor(test_df, dtype=torch.float32)
+val_tensor = torch.tensor(val_df, dtype=torch.float32)
+
+
+# In[5]:
+
+
+output_dir = pathlib.Path("./results").resolve()
+output_dir.mkdir(parents=True, exist_ok=True)
+
+
+# In[6]:
+
+
+# Function to perform dimensionality reduction and extract z matrices for sklearn models
+def extract_z_matrix_sklearn(model, data, model_name, num_components, output_dir):
+    if model_name == "nmf":
+        min_value = data.min().min()
+        if min_value < 0:
+            data = data - min_value
+    # Fit the model and transform the data
+    z_matrix = model.fit_transform(data)
+    
+    # Create a DataFrame and save it
+    z_matrix_df = pd.DataFrame(z_matrix, columns=[f'{model_name}_{i}' for i in range(num_components)])
+    z_matrix_df.insert(0, 'ModelID', train_data.index) 
+    
+    return z_matrix_df
+
+
+# In[7]:
+
+
+def extract_weights(model, model_name, H=None):
+    if model_name in ["pca", "ica", "nmf"]:
+        # Transpose PCA components and format columns
+        weights_df = pd.DataFrame(model.components_, columns=dependency_df.drop(columns=["ModelID"]).columns.tolist()).transpose()
+        weights_df.columns = [f"{x}" for x in range(0, weights_df.shape[1])]
+        
+    # Reset index to rename 'index' to 'genes'
+    weights_df = weights_df.reset_index().rename(columns={"index": "genes"})
+    
+    return weights_df
+
+
+def perform_gsea(weights_df, model_name, num_components):
+    library = blitz.enrichr.get_library("Reactome_2022")
+    seed = random.random()
+    gsea_results = []
+    # Define cutoff values
+    lfc_cutoff = 0.584
+    fdr_cutoff = 0.25
```

Comment:
How did you define these? Also, I recommend making these function arguments

Reply:
I ended up just removing the significance portion as its really not necessary, as the significant dataframe isn't returned